src/main/java/org/terracotta/build/plugins/BuildInfoPlugin.java

```diff
@@ -0,0 +1,71 @@
+package org.terracotta.build.plugins;
+
+import org.gradle.api.Plugin;
+import org.gradle.api.Project;
+import org.gradle.api.provider.Provider;
+import org.gradle.process.internal.ExecException;
+import org.jetbrains.annotations.NotNull;
+import org.terracotta.build.services.Git;
+
+import java.time.Instant;
+import java.util.Objects;
+
+public class BuildInfoPlugin implements Plugin<Project> {
+
+  @Override
+  public void apply(@NotNull Project project) {
+    final Provider<Object> overrideVersion = project.provider(() -> project.findProperty("overrideVersion"));
```

Comment:
There are methods for getting property providers:
```java
final Provider<String> overrideVersion = project.getProviders().gradleProperty("overrideVersion");
```
I think those will work here.